### PR TITLE
add energy indicator

### PR DESCRIPTION
**Pull request by yukinko-iwasaki**
~~Note: Currently asking ITS to white list the API request to fetch the data programatically. Will make this PR ready for a review after I hear back from them.~~

Made a request to OIS to whitelist the url. I am not sure how long this will take. Let's try to merge this now.

### CODE REVIEW

File: energy/energy_generation_consumption.py [hunk 1]
```diff
@@ -4,76 +4,114 @@
...
-    .rename(columns={"Value": "primary_renewable_consumption_share"})
-
-df_total = raw_df[raw_df.Variable == 'Primary Energy Consumption per Capita']
-df_total = df_total.drop(columns=["Unit", "Variable"])\
-    .rename(columns={"Value": "primary_consumption_gigajoules_per_capita"})
+# COMMAND ----------
 
-df = pd.merge(df_renewable, df_total, on=['country_name', 'year'], how='left')
-df
+# Define the fuel types based on the data methodology documentation https://ember-climate.org/data-catalogue/yearly-electricity-data/|
```
Comment (yukinko-iwasaki) [inline energy/energy_generation_consumption.py:32]: These types are based on the data documentation 
image

File: energy/energy_generation_consumption.py [hunk 2]
```diff
@@ -4,76 +4,114 @@
 
 # COMMAND ----------
 
-# Source prevents programmatic data fetch: "https://www.energyinst.org/__data/assets/file/0015/1421601/2023-Country-Transition-Tracker-Consolidated-Format.csv"
-current_dir = Path().resolve()
-file_path = current_dir / '2023-Country-Transition-Tracker-Consolidated-Format.csv'
+# Source prevents programmatic data fetch: https://ember-climate.org/data/api/
+# Curretnly requesting ITS to whitelist this request
+
+file_path = "/Workspace/Users/ysuzuki2@worldbank.org/mega-indicators/energy/yearly_full_release_long_format.csv"
```
Comment (weilu) [inline energy/energy_generation_consumption.py]: Let's update this to point to github location instead, ie 'https://raw.githubusercontent.com/weilu/mega-indicators/main/energy/2023-Country-Transition-Tracker-Consolidated-Format.csv' The link won't work until the PR is merged.

File: energy/energy_generation_consumption.py [hunk 3]
```diff
@@ -4,76 +4,114 @@
 
 # COMMAND ----------
 
-# Source prevents programmatic data fetch: "https://www.energyinst.org/__data/assets/file/0015/1421601/2023-Country-Transition-Tracker-Consolidated-Format.csv"
-current_dir = Path().resolve()
-file_path = current_dir / '2023-Country-Transition-Tracker-Consolidated-Format.csv'
+# Source prevents programmatic data fetch: https://ember-climate.org/data/api/
```
Comment (weilu) [inline energy/energy_generation_consumption.py]: Not source that prevented programmatic data fetch, but rather corporate firewall settings. Let's update the comment to reflect it.

**Review by weilu — APPROVED** [review]
@yukinko-iwasaki this change set looks good to me, with only some minor comments. Feel free to review and merge.

**Pull request merged**